packages/esm-ward-app/src/ward-patient-card/ward-patient-card-row.resources.tsx

```diff
@@ -1,100 +1,75 @@
-      const appliedTo = cardDef.appliedTo;
-
-      return appliedTo == null || appliedTo.some((criteria) => criteria.location == locationUuid);
     });
 
-    const ret = cardDefinition.rows.map((row) => {
-      const { rowType, elements } = row;
-      const patientCardElements = elements.map((patientCardElementId) => {
-        const slot = patientCardElementsMap.get(patientCardElementId);
-        return slot;
-      });
-
-      const WardPatientCardRow: React.FC<WardPatientCardProps> = (props) => {
-        return (
-          <div className={styles.wardPatientCardRow + ' ' + (rowType == 'header' ? styles.wardPatientCardHeader : '')}>
-            {patientCardElements.map((PatientCardElement, i) => (
-              <PatientCardElement {...props} key={i} />
-            ))}
-          </div>
-        );
-      };
+    return cardDefinition;
+  }, [wardPatientCards, locationUuid]);
 
-      return WardPatientCardRow;
-    });
-    return ret;
-  }, [location, wardPatientCards]);
+  if (!currentWardCardConfig) {
+    console.warn(
+      'No ward card configuration has `appliedTo` criteria that matches the current location. Using the default configuration.',
+    );
+    return defaultWardPatientCard;
+  }
 
-  return patientCardRows;
+  return currentWardCardConfig;
 }
 
-function getPatientCardElementFromDefinition(
-  patientCardElementDef: PatientCardElementDefinition,
-): WardPatientCardElement {
-  const { elementType, config } = patientCardElementDef;
-  switch (elementType) {
-    case 'bed-number':
-      return WardPatientBedNumber;
-    case 'patient-name':
-      return WardPatientName;
+export const WardPatientCardElement: React.FC<{ elementId: string; patient: Patient; visit: Visit }> = ({
+  elementId,
+  patient,
+  visit,
+}) => {
+  const { obsElementDefinitions, identifierElementDefinitions } = useConfig<WardConfigObject>().wardPatientCards;
+  const { t } = useTranslation();
+
+  switch (elementId) {
     case 'patient-age':
-      return WardPatientAge;
-    case 'patient-address': {
-      return wardPatientAddress(config.address);
-    }
-    case 'patient-obs': {
-      return wardPatientObs(config.obs);
-    }
+      return <WardPatientAge patient={patient} />;
+    case 'patient-address':
+      return <WardPatientAddress patient={patient} />;
     case 'patient-identifier':
-      return wardPatientIdentifier(config);
-    case 'patient-coded-obs-tags': {
-      return wardPatientCodedObsTags(config.codedObsTags);
+      return <WardPatientIdentifier patient={patient} />;
+    default: {
+      const obsConfig = obsElementDefinitions.find((elementDef) => elementDef.id === elementId);
+      const idConfig = identifierElementDefinitions.find((elementDef) => elementDef.id === elementId);
+      if (obsConfig) {
+        return <WardPatientObs patient={patient} visit={visit} config={obsConfig} />;
+      } else if (idConfig) {
+        return <WardPatientIdentifier patient={patient} config={idConfig} />;
```

Comment:
It feels off to have `WardPatientIdentifier` appear twice in this switch statement, one for having it take the default config and one for having it take a custom config.

I think the `id` field of `obsElementDefinition` / `identifierElementDefinition` is supposed to specify which element it configures. Can we have the `id` be nullable, in which case it applies the config to the default element (patient-identifier / patient-obs)? I think this should simplify the logic here.

Reply:
No, the config system doesn't support types being nullable. `null` can be provided as a default value, but an element with type `Type.String` cannot have `null` provided as the value by configuration.

I'm not sure what feels off about this, but it seems like a more intuitive API to provide than one in which there is a magic value of `id` that modifies the default identifier element.

Reply:
Oh I think I figured out a slightly better way about it